src/party.gleam

```diff
@@ -110,6 +125,25 @@ pub fn either(p: Parser(a, e), q: Parser(a, e)) -> Parser(a, e) {
   Parser(fn(source, pos) { result.or(run(p, source, pos), run(q, source, pos)) })
 }
 
+/// Parse `open`, followed by `p` and `close`. Returns the value returned by `p`.
+/// The values returned by `open` and `close` are discarded.
+pub fn between(
+  open: Parser(_, e),
+  p: Parser(a, e),
+  close: Parser(_, e),
+) -> Parser(a, e) {
+  use _ <- do(open)
+  use x <- do(p)
+  use _ <- do(close)
+  return(x)
+}
+
+/// Parse a line of characters as a String. The new line at the end is discarded
```

Comment:
I would rewrite `new line` to `newline` in the doc comment, since this phrasing a slightly confusing.

Also, much more significantly, it's probably better to return a `List(String)` instead of doing the `map(string.concat)` ourselves, since `map(string.concat)` has a very significant performance penalty and there are many situations (namely single-line comments) where one would want to use `line()` without caring about the parsed string at all, and therefore not want that performance hit. It's easy enough for the user to do `|> party.map(string.concat)` themselves if they need to, and if it comes up enough then we can add a dedicated `until_concat` like I've done with `many_concat` and `many1_concat`. Though that seems unnecessary at the moment.

Reply:
I'll add the line_concat just to stay consistent with the rest of the api